tests/unit/acquisition/test_function.py

```diff
@@ -461,6 +471,175 @@ def prepare_acquisition_function(
     npt.assert_allclose(eci(x), ei(x) * pof(x))
 
 
+class _PseudoTrainableQuadratic(QuadraticMeanAndRBFKernel, PseudoTrainableProbModel):
+    pass
+
+
+class _PseudoTrainableLinear(GaussianProcess, PseudoTrainableProbModel):
+    def __init__(
+        self,
+        *,
+        kernel_amplitude: float | TensorType | None = None,
+    ):
+        kernel = tfp.math.psd_kernels.ExponentiatedQuadratic(kernel_amplitude)
+        super().__init__([lambda x: tf.reduce_sum(x, axis=-1, keepdims=True)], [kernel])
+
+    def __repr__(self) -> str:
+        return "_PseudoTrainableLinear()"
+
+
+_mo_test_models = (
+    _PseudoTrainableQuadratic,
+    _PseudoTrainableLinear,
+    _PseudoTrainableQuadratic,
+)
+
+
+def test_ehvi_builder_raises_for_empty_data() -> None:
+    num_obj = 3
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
```

Comment:
couple of points
  - `empty_dataset` is a helper for this kind of thing. Can save a few key presses.
  - also output shapes don't match
```suggestion
    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, num_obj]))
```